fixing typos in comments and README

alexandear opened the pull request:
The PR corrects typos:

- slack -> Slack
- langauge -> language
- preceed -> precedes
- tha -> that
- mached -> matched
- unit -> uint

Fixes #257 
Fixes #258

ardan-bkennedy commented:
THANK YOU for this. Do you think we can add the corrects from the two issues that are open. Essentially where unsigned int is labeled as `unit` instead of `unit`?

alexandear commented:
@ardan-bkennedy done

> Do you think we can add the corrections from the two issues that are open

Yes, I fixed by replacing `unit` with `uint` in the sentence `When you declare a type
using a non-precision based type (uint, int)`.


> Can you also make sure your name is in the CONTRIBUTORS File.

Added.
